bookmarks/views.py

```diff
@@ -84,20 +84,25 @@ def delete_bookmark_category(request, category_id):
         return HttpResponseRedirect(next)
     else:
         return HttpResponseRedirect(reverse("bookmarks-for-user", args=[request.user.username]))
-    
+
+@login_required    
 @transaction.atomic()
 def download_bookmark_category(request, category_id):
     category = get_object_or_404(BookmarkCategory, id=category_id)
-    licenses_url = (reverse('bookmark-category-licenses', args=[category_id]))
-    #missing: cache checking done in packdownload
-    return download_sounds(licenses_url, category)
+    licenses_content = category.get_attribution()
+    licenses_url = (reverse('category-licenses', args=[category_id]))
+    
+    bookmarked_sounds = Bookmark.objects.filter(category_id=category.id).values("sound_id")
+    sounds_list = Sound.objects.filter(id__in=bookmarked_sounds, processing_state="OK", moderation_state="OK").select_related('user','license')
```

Comment:
Same comment as before, try to do it in a way in which this query is only performed once.